### PR TITLE
chore: update golangci-lint to v1.30.0

### LINKED ISSUE #281: make test never exit
I was trying to run the test locally. When removing `golangci-lint run --deadline=5m`, it passed.

Output of `golangci-lint run --deadline=5m -v`
```
INFO [config_reader] Config search paths: [./ /Users/olsynt/workspace/personal/secrets-store-csi-driver /Users/olsynt/workspace/personal /Users/olsynt/workspace /Users/olsynt /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO Gocritic enabled checks: [appendAssign assignOp caseOrder dupArg dupBranchBody dupCase flagDeref ifElseChain regexpMust singleCaseSwitch sloppyLen switchTrue typeSwitchVar underef unlambda unslice defaultCaseOrder]
INFO [lintersdb] Active 7 linters: [gofmt goimports golint govet ineffassign megacheck.{unused,gosimple} misspell]
INFO [loader] Go packages loading at mode load deps types and syntax took 7.583031758s
INFO [loader] SSA repr building timing: packages building 17.28µs, total 19.949µs
INFO [loader] SSA for megacheck repr building timing: packages building 3.166µs, total 3.495µs
INFO [loader] Packages that do not compile: [sigs.k8s.io/secrets-store-csi-driver/apis/v1alpha1 sigs.k8s.io/secrets-store-csi-driver/pkg/metrics sigs.k8s.io/secrets-store-csi-driver/cmd/secrets-store-csi-driver sigs.k8s.io/secrets-store-csi-driver/controllers [sigs.k8s.io/secrets-store-csi-driver/controllers.test] sigs.k8s.io/secrets-store-csi-driver/pkg/csi-common [sigs.k8s.io/secrets-store-csi-driver/pkg/csi-common.test] sigs.k8s.io/secrets-store-csi-driver/pkg/secrets-store [sigs.k8s.io/secrets-store-csi-driver/pkg/secrets-store.test] sigs.k8s.io/secrets-store-csi-driver/pkg/version [sigs.k8s.io/secrets-store-csi-driver/pkg/version.test] sigs.k8s.io/secrets-store-csi-driver/test/sanity [sigs.k8s.io/secrets-store-csi-driver/test/sanity.test]]
INFO [runner] worker.3 took 112.566663ms with stages: misspell: 58.815493ms, gofmt: 52.843649ms
INFO [runner] worker.2 took 193.561078ms with stages: goimports: 170.128005ms, ineffassign: 23.40315ms, govet: 8.416µs
INFO [runner] worker.4 took 604.544493ms with stages: golint: 604.520047ms
INFO Memory: 1009 samples, avg is 12538.5MB, max is 18001.5MB
INFO Execution took 5m0.02068281s
^C^C^C^C^C 
```

**What steps did you take and what happened:**
```
  git clone ... && cd secrets-store-csi-driver
  make test
```

**What did you expect to happen:**
All test should pass

**Environment:**

- MacOS 10.15.6 
- go version go1.14.6 darwin/amd64

Updating the golangci-lint to latest version helped.

### PR DESCRIPTION
**Pull request by olegsu**
**What this PR does / why we need it**:
`make test` never exit

**Which issue(s) this PR fixes**:
Fixes #281

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,7 +72,7 @@ setup: clean
 	$Q go env
 
 ifndef HAS_GOLANGCI
-	curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(GOPATH)/bin v1.19.1
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin v1.30.0
```
Comment (aramase) [inline Makefile:75]: It looks like the `--deadline` flag has been replaced with `--timeout` in the newer version - https://github.com/golangci/golangci-lint/pull/793. Could you also update that as part of this PR?

Comment (olegsu) [reply Makefile:75]: Done, updated also in `.golangci.yml` file

**Pull request merged**